vector/v.neighbors/main.c

```diff
@@ -54,125 +103,281 @@ int main(int argc, char *argv[])
+
+    if (strcmp(method_opt->answer, "count") != 0) {
+        Fi = Vect_get_field(&In, field);
+        if (Fi == NULL)
+            G_fatal_error(_("Database connection not defined for layer %d"),
+                          field);
+
+        driver = db_start_driver_open_database(Fi->driver, Fi->database);
+        if (driver == NULL)
+            G_fatal_error(_("Unable to open database <%s> by driver <%s>"),
+                          Fi->database, Fi->driver);
+
+        /* check if point column exists */
+        db_get_column(driver, Fi->table, column_opt->answer, &column);
+        if (column) {
+            db_free_column(column);
+            column = NULL;
+        }
+        else {
+            G_fatal_error(_("Column <%s> not found in table <%s>"),
+                          column_opt->answer, Fi->table);
+        }
+
+        /* Check column type */
+        ctype = db_column_Ctype(driver, Fi->table, column_opt->answer);
+
+        if (ctype != DB_C_TYPE_INT && ctype != DB_C_TYPE_DOUBLE)
+            G_fatal_error(_("points_column <%s> of points vector <%s> must be numeric"),
+                          column_opt->answer, Fi->table);
+
+        db_CatValArray_init(&cvarr);
+        nrec = db_select_CatValArray(driver, Fi->table, Fi->key,
+                                     column_opt->answer, NULL, &cvarr);
+        G_debug(1, "selected values = %d", nrec);
+
+        db_close_database_shutdown_driver(driver);
+
+
+    }
 
     G_get_set_window(&region);
     Vect_get_map_box(&In, &box);
-    
+
     if (box.N > region.north + radius || box.S < region.south - radius ||
         box.E > region.east + radius || box.W < region.west - radius) {
-	Vect_close(&In);
-	G_warning(_("Input vector and computational region do not overlap"));
-    }
+			if (box.S > region.north + radius || box.N < region.south - radius ||
+				box.W > region.east + radius || box.E < region.west - radius) 
+					G_fatal_error(_("All points fall outside of the current computational region"));
+        G_warning(_("Input vector and computational region do not overlap"));
+	}
+    
 
     dia = sqrt(region.ns_res * region.ns_res + region.ew_res * region.ew_res);
     if (radius * 2.0 < dia) {
-	G_warning(_("The search diameter %g is smaller than cell diagonal %g: some points could not be detected"),
-	          radius * 2, dia);
+        G_warning(_("The search diameter %g is smaller than cell diagonal %g: some points could not be detected"),
+                  radius * 2, dia);
     }
 
     nrows = Rast_window_rows();
     ncols = Rast_window_cols();
 
-    result = Rast_allocate_buf(CELL_TYPE);
+    result = Rast_allocate_buf(DCELL_TYPE);
     Points = Vect_new_line_struct();
     Cats = Vect_new_cats_struct();
     List = Vect_new_boxlist(0);
 
     /*open the new cellfile */
-    out_fd = Rast_open_new(out_opt->answer, CELL_TYPE);
+    out_fd = Rast_open_new(out_opt->answer, DCELL_TYPE);
```

Comment:
The output raster map does not need to be of type DCELL_TYPE. OTOH, implementing some logic to determine the appropriate output raster datatype might not be trivial, it might depend on both the data type of the column to be used and the selected method.

Reply:
> The output raster map does not need to be of type DCELL_TYPE. OTOH, implementing some logic to determine the appropriate output raster datatype might not be trivial, it might depend on both the data type of the column to be used and the selected method.

@glynnc developed some solution for r.neighbors that might be applicable) See: https://trac.osgeo.org/grass/ticket/2321

Reply:
I've just added code inspired by r.neighbors and others to determine the output raster type.